src/main/java/controller/MainController.java

```diff
@@ -1,241 +1,331 @@
+			File selectedFile = jfc.getSelectedFile();
+			path = selectedFile.getAbsolutePath();
+
+			if (isValid(path)) {
+				uploadFile.close();
+				initMainFrame();
+			} else {
+				uploadFile.displayErrorMessage("File selected is not a valid Excel format!");
+			}
+		}
+	}
+
+	/**
+	 * Initialise the MainFrame and support Frames. Create necessary objects to
+	 * support it.
+	 */
+	private void initMainFrame() {
+		ei = new ExcelImporter(path);
+		excelRows = ei.getAllRows();
+		convertExcelRows();
+		gui = new MainFrame(createExcelTable(), rulesList);
+		qualityGui = new QualityRulesResultFrame();
+		gui.getCheckQualityButton().addActionListener(e -> {
+
+			checkCodeQualityAndShow();
+
+		});
+
+		editButton(this.gui.getEditButton(), this.gui.getRulesComboBox());
+		addButton(this.gui.getAddButton());
+	}
+
+	/**
+	 * Formats all data to a valid format to a JTable and returns a JTable with the
+	 * cell's content
+	 * 
+	 * @return JTable
+	 */
+	private JTable createExcelTable() {
+		String[][] dataForTable = new String[excelRows.size() - 1][excelRows.get(1).length];
+
+		for (int i = 0; i < dataForTable.length; i++) {
+			for (int j = 0; j < dataForTable[i].length; j++) {
+				dataForTable[i][j] = excelRows.get(i + 1)[j].toString();
+			}
+		}
+
+		return new JTable(dataForTable, excelRows.get(0));
+	}
+
+	/**
+	 * Converts all the valid rows into ExcelRow model
+	 * 
+	 * Starts at index 1 because we only want to convert the table content and not
+	 * the header
+	 */
+	private void convertExcelRows() {
+		for (int i = 1; i < excelRows.size(); i++) {
+			try {
+				excelRowsConverted.add(new ExcelRow(excelRows.get(i)));
+			}
+
+			catch (Exception e) {
+				JOptionPane.showMessageDialog(null, "An error occurred while converting the Excel Rows", "Warning",
+						JOptionPane.WARNING_MESSAGE);
+				throw e;
+			}
+		}
+	}
+
+	/**
+	 * This method is used to open the EditRuleController which controls the Rule
+	 * Edition GUI
+	 */
+	private void editButton(JButton editButton, JComboBox<CodeQualityRule> ruleListBox) {
+
+		editButton.addActionListener(e -> {
+
+			CodeQualityRule rule = (CodeQualityRule) ruleListBox.getSelectedItem();
+
+			if (rule == null) {
+				new EditRuleController();
+
+			} else {
+				new EditRuleController(rule);
+			}
+		});
+	}
+
+	/**
+	 * Sets the add Button in the MainFrame to open an empty Rule Edition Popup
+	 */
+	private void addButton(JButton addButton) {
+		addButton.addActionListener(e -> new EditRuleController());
+	}
+
+	/**
+	 * Verify the code quality based on the Rules created and sends the results to
+	 * be displayed in the QualityRulesResultFrame
+	 * 
+	 * @throws ScriptException
+	 */
+	private void checkCodeQualityAndShow() {
+		String[][] results = null;
+		results = getCodeQualityResults();
+		String[] colNames = new String[5 + rulesList.size()];
+		colNames[0] = "Method ID";
+		colNames[1] = "long_method";
+		colNames[2] = "feature_envy";
+		colNames[3] = "PMD";
+		colNames[4] = "iPlasma";

```

Comment:
why are we hardcoding this? can't we get them from the real excel?

Reply:
We could, but we'd have to hardcode the sheet, row and column number of the cell where we'd get this information from. If that's okay, I'll change it around.